src/dev/run_check_core_api_changes.ts

```diff
@@ -0,0 +1,201 @@
+  // Because of the internals of api-extractor ILogger can't be implemented as a typescript Class
+  const warnings = [] as string[];
+  const errors = [] as string[];
+
+  const memoryLogger = {
+    logVerbose(message: string) {
+      return null;
+    },
+
+    logInfo(message: string) {
+      return null;
+    },
+
+    logWarning(message: string) {
+      warnings.push(message);
+    },
+
+    logError(message: string) {
+      errors.push(message);
+    },
+  };
+
+  const options = {
+    // Indicates that API Extractor is running as part of a local build,
+    // e.g. on developer's machine. For example, if the *.api.ts output file
+    // has differences, it will be automatically overwritten for a
+    // local build, whereas this should report an error for a production build.
+    localBuild: acceptChanges,
+    customLogger: memoryLogger,
+  };
+
+  const extractor = new Extractor(apiExtractorConfig, options);
+  extractor.processProject();
+
+  const printableWarnings = warnings.filter(msg => !isApiChangedWarning(msg));
+  const apiChanged = warnings.some(isApiChangedWarning);
+
+  return { apiChanged, warnings: printableWarnings, errors };
+};
+
+(async function run() {
+  const log = new ToolingLog({
+    level: 'info',
+    writeTo: process.stdout,
+  });
+
+  const extraFlags: string[] = [];
+  const opts = getopts(process.argv.slice(2), {
+    boolean: ['accept', 'docs', 'help'],
+    default: {
+      project: undefined,
+    },
+    unknown(name) {
+      extraFlags.push(name);
+      return false;
+    },
+  });
+
+  if (extraFlags.length) {
+    for (const flag of extraFlags) {
+      log.error(`Unknown flag: ${flag}`);
+    }
+
+    process.exitCode = 1;
+    opts.help = true;
+  }
+
+  if (opts.help) {
+    process.stdout.write(
+      dedent(chalk`
+        {dim usage:} node scripts/check_core_api_changes [...options]
+
+        Checks for any changes to the Kibana Core API
+
+        Examples:
+
+          {dim # Checks for any changes to the Kibana Core API}
+          {dim $} node scripts/check_core_api_changes
+
+          {dim # Checks for any changes to the Kibana Core API and updates the documentation}
+          {dim $} node scripts/check_core_api_changes --docs
+
+          {dim # Checks for and automatically accepts and updates documentation for any changes to the Kibana Core API}
+          {dim $} node scripts/check_core_api_changes --accept
+
+        Options:
+          --accept    {dim Accepts all changes by updating the API Review file \`common/core_api_review/kibana.api.ts\` and documentation}
+          --docs      {dim Updates the Core API documentation}
+          --help      {dim Show this message}
+      `)
+    );
+    process.stdout.write('\n');
+    process.exit();
+  }
+
+  log.info('Kibana Core API: checking for changes in API signature...');
+
+  await runBuildTypes();
+
+  const { apiChanged, warnings, errors } = runApiExtractor(opts.accept);
+
+  if (apiChanged) {
```

Comment:
The reason behind the late panic was that I wanted to exit with a non-zero exit when there's warnings so that tests fail, but apart from tests, it's more useful to generate artifacts as far as possible. For instance, forgetting to add an `@public` or `@private` tag should fail tests, but it's still nice to see the documentation output while you're developing so you can see how your changes get "rendered".

Having said that, I'll take a look at this to see if return statements could simplify the if..else logic.